File: linked-lists/Python/reverse_linked_list.py
class ListNode:
    def __init__(self, val=0, next=None):
        self.val = val
        self.next = next


class Solution:
    def reverseList(self, head: ListNode[ListNode]) -> ListNode[ListNode]:
        temp, prev = head, None
        while (temp):
            nxt = temp.next
            temp.next = prev
            prev = temp
            temp = nxt
        return prev

File: linked-lists/Cpp_C/reverse_linked_list.c
#include <string.h>

  struct ListNode {
      int val;
      struct ListNode *next;
  };
struct ListNode* reverseList(struct ListNode* head) {
    struct ListNode* prev = NULL;
    struct ListNode* nxt = NULL;
    while(head) {
        nxt = head->next;
        head->next = prev;
        prev = head;
        head = nxt;
    }
    head = prev;
    return head;
}
